### PR TITLE
Alert workaround

### PR DESCRIPTION
**Pull request by itsbriany**

### CODE REVIEW

**Comment by itsbriany** [comment]
Nooo, I just added a commit and all the previous commit history came back ;(

File: src/rabbitmanager.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ class RabbitManager {
 
   async deleteUserPermissions(user, vhost='/') {
     vhost = encodeURIComponent(vhost);
-    await this.request(`permissions/${vhost}/${user}`, {method: 'DELETE'});
+    await this.request(`permissions/${vhost}/${user}`, {method: 'delete'});
```
Comment (imbstack) [inline src/rabbitmanager.js]: Why this change?

File: src/rabbitmanager.js [hunk 2]
```diff
@@ -158,6 +158,17 @@ class RabbitManager {
     const uriEncodedComponents = this.encodeURIComponents({name: name, vhost: vhost});
     return await this.request(`queues/${uriEncodedComponents.vhost}/${uriEncodedComponents.name}`, {method: 'delete'});
   }
+
+  async messagesFromQueue(queueName, options={count: 5, requeue: true, encoding:'auto'}, vhost='/') {
```
Comment (imbstack) [inline src/rabbitmanager.js]: Perhaps add a http://usejsdoc.org/ comment to this function telling us what it does? It's not obvious from the name.

File: src/rabbitstressor.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/**
```
Comment (imbstack) [inline src/rabbitstressor.js]: I would put this file in a `bin/` directory and not in `src/` since it's not really part of the functioning of the webapp. Then you can remove it from `main.js` as well and just make it directly callable via `node bin/stressor.js` or something of the sort.

**Comment by imbstack** [comment]
Looking good so far!

File: src/rabbitmanager.js (hunk 1 above)
Comment (itsbriany) [inline src/rabbitmanager.js]: I made that change for consistency, since all of the other REST verbs are spelt in lowercase. Either way, I have not seen uppercase or lowercase impact performance or functionality.

File: src/rabbitstressor.js (hunk 3 above)
Comment (itsbriany) [inline src/rabbitstressor.js]: So would it be better to rename the file as `stressor.js` or `queuestressor.js`?
Also, would it still make sense to keep the stressor tests under `test/`?

Comment (itsbriany) [inline src/rabbitstressor.js]: More specifically, if the stressor is moved to `bin/`, it will no longer get executed by the unit tests. Would it make sense to compile the code under `bin/` over to `lib/` as well? I am trying to figure out the best place for the stressor because it is just a convenience tool to add custom stress to queues. Eventually, the plan is to have it go under its own repo from my understanding. But for the moment, I think it needs a temporary home with tests :)

Comment (itsbriany) [inline src/rabbitstressor.js]: So I managed to get the stressor tests running when `rabbitstressor.js` is in the `bin/` directory. However, it looks like [babel-compile](https://www.npmjs.com/package/babel-compile) only compiles files from one directory to another. I.e. it only supports one-to-one mapping. Therefore, in order to use `rabbitstressor.js`, I am importing from a `.bin/` directory (notice this is a hidden directory, just like `./test`). the require syntax looks a bit off. You can see what I mean in the latest commit.

**Comment by imbstack** [comment]
Looks like tests are still failing on the PR travis branch :/

**Comment by itsbriany** [comment]
So I looked deeper into the problem and it looks like there is a race condition where messages are being uploaded and pulled at the same time from the same queue (this test always passes on my machine :)). Do the PR and PUSH travis branches run simultaneously? If they do, do they share the same RabbitMQ service? If so, that could potentially be a reason for why one of the travis tests fail.

**Comment by itsbriany** [comment]
So I made the tests less strict to get them to pass. The goal of the stressor is just to add stress to the queues such that we can collect some stats (i.e. upload a bunch of messages). For now, I do not think it is too important to strictly check the content of the uploaded messages, or how many messages were uploaded (since many machines may upload to the same queue, causing a race condition when reading from the queue). As long as _some_ messages were uploaded, I think we should be OK.

**Comment by imbstack** [comment]
Travis is currently limited to running one job at once so they shouldn't have been stepping on top of one another. Making the tests less strict is probably fine. I'll read the PR now and have more updates shortly.
